tokens/usecase/tokens_usecase.go

```diff
@@ -371,3 +389,22 @@ func (t *tokensUseCase) IsValidChainDenom(chainDenom string) bool {
 	metaData, ok := t.tokenMetadataByChainDenom[chainDenom]
 	return ok && !metaData.IsUnlisted
 }
+
+// IsValidPricingSource implements mvc.TokensUsecase.
+func (t *tokensUseCase) IsValidPricingSource(pricingSource int) bool {
+	switch domain.PricingSourceType(pricingSource) {
+	case domain.ChainPricingSourceType, domain.CoinGeckoPricingSourceType:
+		return true
+	default:
+		return false
+	}
+}
+
+// Get Coingecko ID by chain denom
```

Comment:
nit: for style consistency
```suggestion
// GetCoingeckoIdByChainDenom gets Coingecko ID by chain denom
```